src/OmniSharp.Cake/Extensions/ResponseExtensions.cs

```diff
@@ -85,6 +86,44 @@ public static Task<QuickFixResponse> TranslateAsync(this QuickFixResponse respon
             return response;
         }
 
+        public static async Task<RunCodeActionResponse> TranslateAsync(this RunCodeActionResponse response, OmniSharpWorkspace workspace)
+        {
+            if (response?.Changes == null)
+            {
+                return response;
+            }
+
+            var fileOperations = new List<FileOperationResponse>();
+            var changes = new Dictionary<string, List<LinePositionSpanTextChange>>();
+
+            foreach (var fileOperation in response.Changes)
+            {
+                if (fileOperation.ModificationType == FileModificationType.Modified &&
+                    fileOperation is ModifiedFileResponse modifiedFile)
+                {
+                    await PopulateModificationsAsync(modifiedFile, workspace, changes);
+                }
+
+                fileOperations.Add(fileOperation);
+            }
+
+            foreach (var change in changes)
+            {
+
+                if (!(fileOperations.FirstOrDefault(x => x.FileName == change.Key &&
```

Comment:
You could just invert this to avoid having a `continue` and an ugly `!` before a pattern match. E.g.

```C#
if (fileOperations.FirstOrDefault(x => x.FileName == change.Key &&x.ModificationType == FileModificationType.Modified) is ModifiedFileResponse modifiedFile)
{
    modifiedFile.Changes = change.Value;
}
```

Reply:
Yes, will fix this!